inference-engine/samples/speech_sample/main.cpp

```diff
@@ -404,6 +404,26 @@ void sumPerformanceCounters(std::map<std::string, InferenceEngine::InferenceEngi
     }
 }
 
+std::vector<std::string> ParseScaleFactors(const std::string& str) {
+    std::vector<std::string> scaleFactorInput;
+
+    if (!str.empty()) {
+        std::string outStr;
+        std::istringstream stream(str);
+        int i = 0;
+        while (getline(stream, outStr, ',')) {
+            auto floatScaleFactor  = std::stof(outStr);
+            if (floatScaleFactor <= 0.0f) {
+                throw std::logic_error("Scale factor for " + std::to_string(i) + " input out of range (must be non-negative).");
```

Comment:
I understand this was in the original code, but let's fix the message - it should be "positive", not "non-negative".

Reply:
And i is never incremented